### DISCUSSION TITLE
Correct way to start gnome-keyring-daemon

### DISCUSSION THREAD

**Discussion by cherio** [Xpra-org/xpra#4266] [General]
My original trouble is with google `chrome` launched within xpra on a remote ubuntu workstation. It doesn't seem to be able to connect to `gnome-keyring-daemon`. When chrome is launched within xfce4-session (either local display or NoMachine new session) it pops up a dialog prompting to unlock the keyring which enables access to saved passwords and other chrome features. The keyring unlock popup doesn't show when the same chrome is launched from xpra client.

I do launch the server with dbus session as `--dbus-launch="dbus-launch --close-stderr"` and update DBUS daemon with DISPLAY information `dbus-update-activation-environment DISPLAY` within the opened session.

I wonder if there is a solution or a recommended approach for using gnome keyring in a xpra session.
```shell
$ xpra --version
xpra v6.1-r35816 (gefa5e2cf65) beta
```
```shell
$ uname -a
Linux HOSTNAME 6.5.0-1022-azure # 23~22.04.1-Ubuntu SMP Thu May  9 17:59:24 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by totaam** [comment]
Have you tried `--source=gnome-keyring-daemon` or `--source-start=gnome-keyring-daemon` ?

**Reply by cherio** [reply]
I have not. I saw examples with these options referenced in FAQ but they are not in the `man`; to me, using them is a guesswork akin to practicing alchemy ;)

Specifying `--source='/usr/bin/gnome-keyring-daemon --start --components=secrets'` worked! Thanks!

Is there a reference/description of how these arguments fit into xpra launch sequence? Are "--source/--source-start" arguments executed before "--start" arguments and in what order? Or, are they being executed asynchronously?

Thank you!

**Reply by totaam** [reply]
`xpra --help` has a short description, which should be added to the man page.
More information here: https://github.com/Xpra-org/xpra/issues/2688
